ocaml/xapi/helpers.ml

```diff
@@ -291,6 +291,16 @@ let call_api_functions ~__context f =
          with e ->
            debug "Helpers.call_api_functions failed to logout: %s (ignoring)" (Printexc.to_string e))
 
+let test_mode = ref false
+let call_api_functions ~__context ?test_fn f =
```

Comment:
How about using the presence of `test_fn` to decide about being in a test? When this is an optional argument with default `None` you might not need the reference.